roles/kubernetes/node/templates/kubelet.standard.env.j2

```diff
@@ -73,16 +73,24 @@ KUBELET_HOSTNAME="--hostname-override={{ kube_override_hostname }}"
 {% endif %}
 
 {# Kubelet node labels #}
+{% set role_node_labels = [] %}
 {% if inventory_hostname in groups['kube-master'] %}
-{%   set node_labels %}--node-labels=node-role.kubernetes.io/master=true{% endset %}
+{%   do role_node_labels.append('node-role.kubernetes.io/master=true') %}
 {%   if not standalone_kubelet|bool %}
-{%     set node_labels %}{{ node_labels }},node-role.kubernetes.io/node=true{% endset %}
+{%     do role_node_labels.append('node-role.kubernetes.io/node=true') %}
 {%   endif %}
 {% else %}
-{%   set node_labels %}--node-labels=node-role.kubernetes.io/node=true{% endset %}
+{%   do role_node_labels.append('node-role.kubernetes.io/node=true') %}
 {% endif %}
+{% set inventory_node_labels = [] %}
+{% if node_labels is defined %}
+{% for labelname, labelvalue in node_labels.iteritems() %}
+{% do inventory_node_labels.append(labelname + '=' + labelvalue) %}
+{% endfor %}
+{% endif %}
+{% set all_node_labels = role_node_labels + inventory_node_labels %}
```

Comment:
All nodes are labeled, I feel not very good For the production environment, can  you try to specify the name of the node to label it?

Reply:
I don't understand your question.
Each node in the inventory can have different labels...

Reply:
if node_labels is defined ,All nodes will be marked with the same label.

Reply:
Indeed, if "node_labels" is defined only in group_vars/*.yml all nodes will have the same labels (plus role_node_labels which will be different on nodes and masters).
But the node_labels set in the inventory (e.g. inventory.{cfg,yaml}) will override the ones in group_vars/*.yml so each node will have its own labels